src/screens/TokenDetail.js

```diff
@@ -0,0 +1,685 @@
+import QRCode from 'qrcode.react';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import HathorAlert from '../components/HathorAlert';
+import { connect } from "react-redux";
+import ReactLoading from 'react-loading';
+import hathorLib from '@hathor/wallet-lib';
+
+const mapStateToProps = (state) => {
+  return {
+    historyTransactions: state.historyTransactions,
+  };
+};
+
+
+/**
+ * Screen to manage a token. Mint, melt, edit, unregister, configuration string.
+ *
+ * @memberof Screens
+ */
+class TokenDetail extends React.Component {
+  constructor(props) {
+    super(props);
+
+    /**
+     * token {Object} selected token data
+     * mintOutputs {Object} array with outputs available to mint
+     * meltOutputs {Object} array with outputs available to melt
+     * walletAmount {number} amount available of this token on this wallet
+     * destroyQuantity {number} holds the destroy quantity typed in the input
+     * action {string} selected action (mint, melt, delegate-mint, delegate-melt, destroy-mint, destroy-melt)
+     * successMessage {string} success message to show
+     * errorMessage {string} error message to show
+     * loading {boolean} if should show loading spinner
+     * pin {string} pin typed on input
+     * formValidated {boolean} if form was already validated
+     */
+    this.state = {
+      token: null,
+      mintOutputs: [],
+      meltOutputs: [],
+      walletAmount: 0,
+      destroyQuantity: 0, // this is the only one I need as state because I use this value to show message on the confirm modal
+      action: '',
+      successMessage: '',
+      errorMessage: '',
+      loading: false,
+      pin: '',
+      formValidated: false,
+    };
+
+    // Mint/melt refs
+    this.amount = React.createRef();
+    this.createAnother = React.createRef();
+    this.chooseAddress = React.createRef(); // Only for mint
+    this.address = React.createRef(); // Only for mint
+    this.addressWrapper = React.createRef(); // Only for mint
+
+    // Delegate refs
+    this.delegateAddress = React.createRef();
+    this.delegateCreateAnother = React.createRef();
+
+    // Destroy quantity ref
+    this.destroyQuantity = React.createRef();
+
+    // Form ref
+    this.form = React.createRef();
+  }
+
+  componentDidMount() {
+    const { match: { params } } = this.props;
+
+    const allTokens = hathorLib.tokens.getTokens();
+    const token = allTokens.find((data) => data.uid === params.tokenUID);
+
+    this.setState({ token }, () => {
+      this.updateTokenData();
+    });
+  }
+
+  componentDidUpdate = (prevProps) => {
+    if (this.props.historyTransactions !== prevProps.historyTransactions) {
+      this.updateTokenData();
+    }
+  }
+
+  /**
+   * Update token state after didmount or props update
+   */
+  updateTokenData = () => {
+    const filteredHistoryTransactions = hathorLib.wallet.filterHistoryTransactions(this.props.historyTransactions, this.state.token.uid, true);
```

Comment:
I don't think we need to go through all transactions. We can follow only the ones with at least one authority output, can't we?

Reply:
Yes, that's right. I've updated the code.